### PR TITLE
Document `GET /api/collections` endpoint

### PR DESCRIPTION
**Pull request by Wauplin**
Followup after integration in `huggingface_hub`: https://github.com/huggingface/huggingface_hub/pull/1856

cc @enzostvs to update the [Hub API Playground](https://huggingface.co/spaces/enzostvs/hub-api-playground) :pray: . How do you want to proceed on this btw? Is it fine if I ping you here each time I document a new endpoint?

New section: https://moon-ci-docs.huggingface.co/docs/hub/pr_1137/en/api#get-apicollections

### CODE REVIEW

File: docs/hub/api.md [hunk 1]
```diff
@@ -288,6 +288,32 @@ Return information about a collection.
 
 This is equivalent to `huggingface_hub.get_collection()`.
 
+### GET /api/collections
+
+List collections from the Hub, based on some criteria. The supported parameters are:
+- `owner` (string): filter collections created by a specific user.
+- `item` (string): filter collections containing a specific item. Value must be the item_type and item_id concatenated. Example: `"models/teknium/OpenHermes-2.5-Mistral-7B"`, `"datasets/squad"` or `"papers/2311.12983"`.
+- `sort` (string): sort the returned collections. Supported values are `"lastModified"`, `"trending"` (default) and `"upvotes"`.
+- `limit` (int): maximum number of collections per pages.
```
Comment (Kakulukian) [inline docs/hub/api.md:297]:
```suggestion
- `limit` (int): maximum number (100) of collections per page.
- `q` (string): Filter based on substrings for titles & description
```

**Review by Kakulukian — APPROVED** [review]
thanks :)

**Review by osanseviero — APPROVED** [review]
Thanks!

**Comment by Wauplin** [comment]
Thanks for the reviews @Kakulukian  @osanseviero. I applied the changes and will now merge it.

**Pull request merged**